Sources/SimpleSalesforceSDK/Models/SFAuth.swift

```diff
@@ -57,8 +57,13 @@ class SFAuth {
   /// This will also call `interospectAccessToken` to reset the expiry
   func refreshAccessTokenIfNeeded(config: SFConfig) async throws {
     guard !self.isAccessTokenValid else { return }
-    
-    try! await self.refreshAccessToken(config: config)
+    do {
+      try await self.refreshAccessToken(config: config)
+    } catch {
+      self.reset()
+      print("Error while refreshing the access token...")
+      print(String(describing: error))
```

Comment:
Don’t you want to raise again? So the user of the SDK knows that something went wrong?